src/openforms/config/models.py

```diff
@@ -649,6 +655,33 @@ def as_dict(self):
         return {k: list(v) for k, v in ret.items()}
 
 
+class CSPSettingManager(models.Manager.from_queryset(CSPSettingQuerySet)):
+    def set_for(self, obj: models.Model, settings: list[tuple[str, str]]) -> None:
+        """
+        Deletes all the connected csp settings and creates new ones based on the new provided data.
+        """
+        instances = []
+        for setting in settings:
+            directive, value = setting
+            if directive not in CSPDirective.values:
+                logger.error(
+                    "Could not create csp setting for model '%s'. '%s' is not a valid directive.",
+                    obj,
+                    directive,
+                )
+                return
+
+            instances.append(
+                CSPSetting(content_object=obj, directive=directive, value=value)
+            )
```

Comment:
```suggestion
class CSPSettingManager(models.Manager.from_queryset(CSPSettingQuerySet)):
    def set_for(self, obj: models.Model, settings: list[tuple[CSPDirective, str]]) -> None:
        """
        Deletes all the connected csp settings and creates new ones based on the new provided data.
        """
        instances = [
            CSPSetting(content_object=obj, directive=directive, value=value)
            for directive, value in settings
        ]
```
A couple things here to remark:

1. If you update the type hint from str -> CSPDirective, you force callers to use the `CSPDirective.foo` constants (at a type checking level). Relying on the type checker then allows you to drop the defensive programming clause to reject "invalid" values.
2. you can immediately destructure the `setting` tuple in the for loop
3. dropping the entire guard clause turns this into a trivial list comprehension.